Content.Server/Resist/ResistLockerSystem.cs

```diff
@@ -0,0 +1,77 @@
+using Content.Shared.Movement;
+using Robust.Shared.GameObjects;
+using Content.Server.Storage.Components;
+using Content.Server.DoAfter;
+using Content.Shared.Popups;
+using Robust.Shared.Localization;
+
+namespace Content.Server.Resist.EntitySystems
+{
+    public class ResistLockerSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ResistLockerComponent, RelayMovementEntityEvent>(OnRelayMovement);
+        }
+
+
+        private void OnRelayMovement(EntityUid uid, ResistLockerComponent component, RelayMovementEntityEvent args)
+        {
+            if(EntityManager.TryGetComponent<EntityStorageComponent?>(uid, out var storageComponent))
+            {
+                if (!component.IsResisting)
+                {
+                    //Check IfWeldedShut or if Lock exists and is locked
+                    if (EntityManager.TryGetComponent<LockComponent?>(uid, out var lockComponent) && lockComponent.Locked || storageComponent.IsWeldedShut)
```

Comment:
the generic argument doesn't need to be nullable, also might want to use the shorter `TryComp` as opposed to `EntityManager.TryGetComponent`